tcl/envmngt.tcl.in

```diff
@@ -1236,6 +1278,25 @@ ml <- function(...){
 }}
          }
       }
+      pwsh {
+         if {[getState is_win]} {
+            reportErrorAndExit {No autoinit mode available for 'pwsh' shell on Windows}
+         }
+
+         set fdef "function global:envmodule {"
```

Comment:
Return codes of functions are a bit weird in powershell. You can either use `Write-Error` or `throw` in powershell to get the calling shell to recognize that the function failed. I can try to capture the stderr stream from the modulecmd.tcl call and reprint the message via one of the two options. The only problem is that this will always add something to the error message:

`Write-Error`:
```
envmodule: ERROR: Invalid command 'foo'
  Try 'module --help' for more information.
```

`throw`:
```
Exception: ERROR: Invalid command 'foo'
  Try 'module --help' for more information.
```

So we have to account for that in the test suite. I'd vote for the exception since that could be useful if the user want's to catch that via `try {} catch {}`

Reply:
As an alternative, there is the `LastExitCode` variable which we could manually set inside the function. This variable is pretty much independent of the actual execution status `$?`. `$?` is only false if we exit the function via `Write-Error` or `throw`. One place where we don't want that is for example `module is-loaded ...`. If the module in question isn't loaded, I'd say we don't want to throw an exception but rather set the `LastExitCode` or actually return a bool

Reply:
On other shells, it returns `_mlstatus` which acts as a bool. I agree that it should not throw an error on query operations like `is-loaded`.

Reply:
We could always return `_mlstatus` but that means that each command prints either `True`, `False`, or whatever text is stored in it to the console. Powershell acts more like a language like Python than like "normal" shells. I'll see if I can find a better solution.